src/Microsoft.Azure.SignalR/HubHost/ServiceHubDispatcher.cs

```diff
@@ -158,8 +153,17 @@ private async Task GracefulShutdownAsync(GracefulShutdownOptions options, Cancel
         Log.SettingServerOffline(_logger, _hubName);
         await _serviceConnectionManager.OfflineAsync(options.Mode, cancellationToken);
 
-        Log.TriggeringShutdownHooks(_logger, _hubName);
-        await options.OnShutdown(Context);
+        if (options.Mode == GracefulShutdownMode.Off)
+        {
+            // By default we directly close all the client connections
+            Log.CloseClientConnections(_logger, _hubName);
+            await _serviceConnectionManager.CloseClientConnections(cancellationToken);
```

Comment:
`CloseClientConnections` seems to be called in multiple places, are they doing same tasks?

Reply:
I don't think it is called in multiple places, it's passing down from manager to container and finally to the service connection